docs/Meadow/Meadow.Foundation/Working_with_Sensors/Sensor_Service/index.md

```diff
@@ -0,0 +1,36 @@
+---
+layout: Meadow
+title: Optimize Sensor Reads with SensorService
+subtitle: Reduce sensor thread usage by isolating reads to a the shared SensorService.
+---
+
+While developing code to read sensors asyncronously in C#, it can be very easy to create a large number of threads because of how simple async Task-based code is to write. On low-resource platforms like Meadow, too many threads can cause excessive CPU cycles as it switches between contexts. To help with this, Meadow provides the `SensorService` to help avoid this when you are regularly reading for several sensors every second or more.
+
+## What does SensorService provide?
+
+The `SensorService` provides a shared sensor reading thread that can be used to poll sensors at a regular interval. This allows you to avoid creating a new thread for each sensor, which can be resource-intensive on low-resource platforms like Meadow.
+
+Once you have registered your polling sensors with the `SensorService`, you continue to read values from them using their respective `Read` and events. Behind the scenes, the `SensorService` will manage the polling of the sensors at the regular interval using the shared thread.
+
+`SensorService` is transparent to the sensor. It doesn't change how you interact with the sensor, it just manages the polling of the sensor, replacing the timer with the timer built into the service.
```

Comment:
"...replacing the timer..." -> "...replacing the sensor's internal thread loop..."

Reply:
Clarified that...thank you!